apps/shared/avifpng.c

```diff
@@ -368,11 +384,33 @@ avifBool avifPNGRead(const char * inputFilename,
     return readResult;
 }
 
+//------------------------------------------------------------------------------
+// Writing
+
+// Does the opposite of avifCopyRawProfile(): writes a payload as a raw profile string.
+static avifBool avifBytesToRawProfile(const avifRWData * bytes, const char * profileName, avifRWData * profile)
```

Comment:
At first I wanted to avoid implementing both, because currently there is no guarantee that the used libpng version has `PNG_eXIf_SUPPORTED` and/or `PNG_iTXt_SUPPORTED`. I added these but I also kept the raw profile fallback if they are not available. There is also the possibility of a `\0` in the given XMP, which is forbidden in `iTXt` (`zTXt` raw profile fallback here too).

The alternative would be requiring `PNG_eXIf_SUPPORTED` and `PNG_iTXt_SUPPORTED` at compile time, and failing PNG exports of XMP with `\0`. Bumping the minimum libpng version would not be so bad I guess? This can be done in another PR.

Reply:
Done in https://github.com/AOMediaCodec/libavif/pull/1193.